pkg/apis/pipeline/v1beta1/task_validation_test.go

```diff
@@ -855,6 +873,72 @@ func TestTaskSpecValidateError(t *testing.T) {
 			Message: `variable is not properly isolated in "not isolated: $(params.baz[*])"`,
 			Paths:   []string{"steps[0].args[0]"},
 		},
+	}, {
+		name: "object used in unaccepted field",
+		fields: fields{
+			Params: []v1beta1.ParamSpec{{
+				Name: "gitrepo",
+				Type: v1beta1.ParamTypeObject,
+				Properties: map[string]v1beta1.PropertySpec{
+					"url":    {},
+					"commit": {},
+				},
+			}},
+			Steps: []v1beta1.Step{{
+				Name:       "do-the-clone",
+				Image:      "some-git-image",
+				Args:       []string{"$(params.gitrepo)"},
+				WorkingDir: "/foo/bar/src/",
+			}},
+		},
+		expectedError: apis.FieldError{
+			Message: `variable type invalid in "$(params.gitrepo)"`,
+			Paths:   []string{"steps[0].args[0]"},
+		},
+	}, {
+		name: "object star used in unaccepted field",
```

Comment:
This is a case for substituted field being an array, and substituting field is an object. Can you also add one for substituted field being a string? "object star used in unaccepted field" also worths changing a name. I think it would be helpful if you can organize the test in such way:
1. object used in array field
2. object used in string field
3. object star used in array field
4. object star used in string field
...
This could help me identifying missing cases easier.

Reply:
sg. done!